test/unit/Identifier/IdentifierTest.php

```diff
@@ -44,4 +47,28 @@ public function testIsTypesForFunction() : void
         self::assertFalse($identifier->isClass());
         self::assertTrue($identifier->isFunction());
     }
+
+    public function testGetNameForClosure() : void
+    {
+        $identifier = new Identifier(ReflectionFunctionAbstract::CLOSURE_NAME, new IdentifierType(IdentifierType::IDENTIFIER_FUNCTION));
+        self::assertSame(ReflectionFunctionAbstract::CLOSURE_NAME, $identifier->getName());
+    }
+
+    public function testGetNameForAnonymousClass() : void
+    {
+        $identifier = new Identifier(ReflectionClass::ANONYMOUS_CLASS_NAME_PREFIX . ' filename.php', new IdentifierType(IdentifierType::IDENTIFIER_CLASS));
+        self::assertStringStartsWith(ReflectionClass::ANONYMOUS_CLASS_NAME_PREFIX, $identifier->getName());
+    }
+
+    public function testGetNameForWildcard() : void
+    {
+        $identifier = new Identifier(Identifier::WILDCARD, new IdentifierType(IdentifierType::IDENTIFIER_CLASS));
+        self::assertSame(Identifier::WILDCARD, $identifier->getName());
+    }
+
+    public function testThrowExceptionForInvalidName() : void
+    {
+        self::expectException(InvalidIdentifierName::class);
+        new Identifier('1234567890', new IdentifierType(IdentifierType::IDENTIFIER_CLASS));
```

Comment:
Move the invalid data to a data provider, try a few invalid strings

Reply:
Done.